File: api-java/src/main/java/fr/huiitre/tools/modules/riot/valorant/application/catalog/usecase/ListValorantWeaponsUseCase.java
package fr.huiitre.tools.modules.riot.valorant.application.catalog.usecase;

import java.util.List;
import java.util.Optional;

import org.springframework.stereotype.Service;

import fr.huiitre.tools.modules.core.module.domain.ModuleCode;
import fr.huiitre.tools.modules.core.role.domain.RoleCode;
import fr.huiitre.tools.modules.core.security.application.usecase.SecuredUseCase;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.ports.ValorantWeaponRepository;
import fr.huiitre.tools.modules.riot.valorant.application.catalog.view.ValorantWeaponView;

@Service
public class ListValorantWeaponsUseCase implements SecuredUseCase {

    private final ValorantWeaponRepository weaponRepository;

    public ListValorantWeaponsUseCase(ValorantWeaponRepository weaponRepository) {
        this.weaponRepository = weaponRepository;
    }

    @Override
    public Optional<ModuleCode> requiredModule() {
        return Optional.of(ModuleCode.RIOT);
    }

    @Override
    public RoleCode requiredRole() {
        return RoleCode.READ_ONLY;
    }

    public List<ValorantWeaponView> execute() {
        return weaponRepository.findAll();
    }
}

File: api/Modules/Riot/Valorant/Application/Catalog/Usecases/ListValorantWeaponsUseCase.cs
using Tools.Api.Modules.Core.Security.Application.Services;
using Tools.Api.Modules.Core.Security.Application.Usecases;
using Tools.Api.Modules.Core.Security.Domain;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Ports;
using Tools.Api.Modules.Riot.Valorant.Application.Catalog.Views;

namespace Tools.Api.Modules.Riot.Valorant.Application.Catalog.Usecases;

public sealed class ListValorantWeaponsUseCase(
    UseCaseAuthorizer authorizer,
    IValorantWeaponRepository weaponRepository
) : SecuredUseCase(authorizer)
{
    protected override RoleCode RequiredRole => RoleCode.ReadOnly;
    protected override ModuleCode? RequiredModule => ModuleCode.Riot;

    public Task<List<ValorantWeaponView>> Execute()
    {
        return weaponRepository.FindAll();
    }
}
